staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -393,24 +425,33 @@ func newInformer(
 		Process: func(obj interface{}) error {
 			// from oldest to newest
 			for _, d := range obj.(Deltas) {
+				obj := d.Object
+				if transformer != nil {
```

Comment:
Have I correctly skimmed the code below and it computes the keys for any indexes *after* the transform, so the data to compute is always present in the returned instances?

Reply:
> and it computes the keys for any indexes after the transform,

Yes.

> so the data to compute is always present in the returned instances?

I guess it depends on the indexes that are passed. But in the correct usage the index should only use data from transformed objects.